functions

```diff
@@ -29,61 +29,93 @@ function _feedback() {
 }
 
 
-function process_security_tool_exit_status() {
+function process_command_exit_status() {
   # Store the provided exit status to return it at the end
   exit_status="${1}"
-  security_tool="${2}"
-  action="${3}"
+  command="${2}"
+  description="${3}"
 
   if [[ "${exit_status}" != 0 ]]; then
-    _feedback ERROR "Failed ${security_tool} ${action}"
+    _feedback ERROR "Failed ${command} ${description}"
   else
-    _feedback INFO "Passed ${security_tool} ${action}"
+    _feedback INFO "Passed ${command} ${description}"
   fi
 
   # If the shell is not interactive ("i" is not in the $- variable), skip
-  # multiple runs of the provided security tool by timestamping the touch file
+  # multiple runs of the provided command by timestamping the touch file
   if [[ ! "${-}" =~ .*i.* ]]; then
-    date +%s > "/${security_tool}_complete"
+    # If there are any spaces in the provided command, replace them with _s
+    date +%s > "/${command// /_}_complete"
   fi
 
-  return ${exit_status}
+  return "${exit_status}"
 }
 
 
 function terraform() {
-  # Methods to skip all security tools for terraform
+  # Methods to skip **all** security tools for terraform
+  DISABLE_SECURITY="${DISABLE_SECURITY:-false}"
+
   if [[ "${1}" == "--disable-security" ]]; then
     _feedback WARNING "Skipping all security checks for terraform due to the --disable-security argument"
     shift
     command "${FUNCNAME[0]}" "$@"
     return $?
-  elif [[ "${DISABLE_SECURITY:-false}" == "true" ]]; then
+  elif [[ "${DISABLE_SECURITY,,}" == "true" ]]; then
     _feedback WARNING "Skipping all security checks for terraform due to the DISABLE_SECURITY environment variable value"
     command "${FUNCNAME[0]}" "$@"
     return $?
+  else
+    ## Validate the input prior to running security tooling
+    # If the initialization was already run successfully in an non-interactive
+    # shell, don't run it again
+    if [[ -r "/terraform_init_complete" ]]; then
+      _feedback INFO "Skipping `terraform init` because it was already run on $(date -d @"$(cat /terraform_init_complete)")"
+    else
+      command terraform init &>/terraform_init_output
+      process_command_exit_status "${?}" "terraform init" "initialization"
+      return=${?}
+      if [[ ${return:-1} != 0 ]]; then
+        cat /terraform_init_output
+        return "${return}"
+      fi
+    fi
+    # If the initialization was already run successfully in an non-interactive
+    # shell, don't run it again
+    if [[ -r "/terraform_validate_complete" ]]; then
+      _feedback INFO "Skipping `terraform validate` because it was already run on $(date -d @"$(cat /terraform_validate_complete)")"
+    else
+      command terraform validate &>/terraform_validate_output
```

Comment:
Similar to the above comments, `terraform_validate_output` -> `terraform_validate_complete`